### PR TITLE
JENKINS-26036 add support for folders and other possible view containers

### PR DESCRIPTION
**Pull request by apemberton**
@reviewbybees

### CODE REVIEW

**Comment by apemberton** [comment]
After further testing nested-view support, I realized this more generic approach allows removing the nested-view dependency altogether.

File: src/main/java/de/pellepelster/jenkins/walldisplay/WallDisplayViewAction.java [hunk 1]
```diff
@@ -8,44 +8,43 @@
 
 /**
  * Action providing the link to the actual walldisplay page
- *
+ * 
  * @author Christian Pelster
  */
 public class WallDisplayViewAction implements Action {
 
-    private static final String ENCODING = "UTF-8";
```
Comment (jglick) [inline src/main/java/de/pellepelster/jenkins/walldisplay/WallDisplayViewAction.java]: Please do not reformat existing code. In general a PR should include the minimum possible number of diff lines.

**Comment by apemberton** [comment]
@jglick agree; this was unintentional.

**Comment by jglick** [comment]
Tip: [this URL](https://github.com/jenkinsci/walldisplay-plugin/pull/42/files?w=1) can be used to see the diff ignoring whitespace.

**Comment by apemberton** [comment]
Very cool; didn't know that option existed.

**Comment by cdemeijer** [comment]
:+1: Would love to see this in a release :)

**Comment by pellepelster** [comment]
me too, i'm struggeling with the f***k maven release plugin but i think i will win the fight within the next days =)

**Comment by cdemeijer** [comment]
I see you won the fight. Great :)
I just grabbed the release and built and deployed it this morning. Building was easier than I expected.
But either I'm doing something wrong (not unlikely) or I misunderstood what the PR should achieve.
I have a view that contains jobs directly and jobs that are inside folders. Should the wall not display both the top level jobs as well as the nested jobs?

On a sidenote: styling does not appear to work on a wall display of a folder.

**Pull request merged**